packages/sitecore-jss-nextjs/src/edge/personalize-middleware.test.ts

```diff
@@ -870,4 +870,86 @@ describe('PersonalizeMiddleware', () => {
       nextRewriteStub.restore();
     });
   });
+
+  describe('middleware connection errors', () => {
+    const req = createRequest();
+    const res = createResponse({
+      body: '<div> Regular page </div>',
+    });
+
+    const errorSpy = spy(console, 'log');
+
+    afterEach(() => {
+      errorSpy.resetHistory();
+    });
+
+    after(() => {
+      errorSpy.restore();
+    });
+
+    it('should log error when executeExperience throws', async () => {
+      const { middleware, executeExperience } = createMiddleware();
+
+      const error = new Error('CDP executeExperience fails');
+
+      executeExperience.restore();
+
+      const executeExperienceWithError = sinon
```

Comment:
@art-alexeyenko I think we should be able to structure them like this so that we don't need to pass `middleware` to the prop functions:

```
const executeExperience = (middleware['cdpService']['executeExperience'] =
  props.executeExperienceStub ?? sinon.stub().returns(Promise.resolve(props.variantId)));
```

Reply:
Done! Please take a look.

Reply:
Very nice! 👍